### PR TITLE
Refactor optional concentration_scaling in ecckd

### PR DESCRIPTION
**Pull request by drieg**
- Reduce the number of `IF (PRESENT(concentration_scaling))` checks
- Introduce concentration_scaling also for the vectorized version `calc_optical_depth_ckd_model_vec`

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf-ifs/ecrad?pullRequest=27) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf-ifs/ecrad?pullRequest=27) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf-ifs/ecrad?pullRequest=27) it.</sub>

File: radiation/radiation_ecckd.F90 [hunk 1]
```diff
@@ -456,7 +456,7 @@ end subroutine read_spectral_solar_cycle
   ! at nlev layers
   subroutine calc_optical_depth_ckd_model(this, ncol, nlev, istartcol, iendcol, nmaxgas, &
        &  pressure_hl, temperature_fl, mole_fraction_fl, &
-       &  optical_depth_fl, rayleigh_od_fl, concentration_scaling)
+       &  optical_depth_fl, rayleigh_od_fl, opt_concentration_scaling)
```
Comment (reuterbal) [inline radiation/radiation_ecckd.F90:459]: I would prefer not to change the interface, i.e., keep the name `concentration_scaling` here - the `opt_` prefix is not used elsewhere for optional arguments either.

**Review by reuterbal — COMMENTED** [review]
Apologies for the delay, conceptually this looks great and I'll test this locally before merging.

File: radiation/radiation_ecckd.F90 (hunk 1 above)
Comment (drieg) [reply radiation/radiation_ecckd.F90:459]: Thanks. Let me know if your local tests were successful, then I can perform this change. Any preferences for the naming of the local variable (currently `concentration_scaling`)?

Comment (reuterbal) [reply radiation/radiation_ecckd.F90:459]: No specific preference, for a local variable we're also not very restricted. Intuitively I might have gone for something like a `local_` prefix, but `opt_` or something else would also be fine.

**Review by reuterbal — COMMENTED** [review]
Many thanks! Changes work as expected. If you could revert the interface change as commented, otherwise ready to merge.

**Comment by drieg** [comment]
I reverted the interface changes as suggested. Thanks!

**Review by reuterbal — APPROVED** [review]
Looks great, many thanks!

**Pull request merged**